File: android/api-tests/src/test/java/com/revenuecat/apitests/java/CustomerInfoMapperApiTests.java
package com.revenuecat.apitests.java.mappers;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.hybridcommon.mappers.CustomerInfoMapperKt;

import java.util.Map;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

@SuppressWarnings("unused")
class CustomerInfoMapperApiTests {
    private void checkMapAsync(CustomerInfo customerInfo, Function1<? super Map<String, ?>, Unit> callback) {
        CustomerInfoMapperKt.mapAsync(customerInfo, callback);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/mappers/CustomerInfoMapperApiTests.kt
package com.revenuecat.apitests.kotlin.mappers

import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.hybridcommon.mappers.mapAsync

@Suppress("unused", "UNUSED_VARIABLE")
private class CustomerInfoMapperApiTests {
    fun checkMapAsync(customerInfo: CustomerInfo, callback: (Map<String, Any?>) -> Unit) {
        val unit: Unit = customerInfo.mapAsync(callback)
    }
}
